lib/elixir/lib/string.ex

```diff
@@ -750,129 +733,248 @@ defmodule String do
-      iex> String.lstrip("   abc  ")
-      "abc  "
+      iex> String.trim_leading("\n  abc   ")
+      "abc   "
 
   """
-  defdelegate lstrip(binary), to: String.Break
+  @spec trim_leading(t) :: t
+  defdelegate trim_leading(string), to: String.Break
 
   @doc """
-  Returns a string where all leading `char`s have been removed.
+  Returns a string where all leading `to_trim`s have been removed.
 
   ## Examples
 
-      iex> String.lstrip("_  abc  _", ?_)
-      "  abc  _"
+      iex> String.trim_leading("__ abc _", "_")
+      " abc _"
 
-  """
-  @spec lstrip(t, char) :: t
-  def lstrip(string, char)
+      iex> String.trim_leading("1 abc", "11")
+      "1 abc"
 
-  def lstrip(<<char::utf8, rest::binary>>, char) when is_integer(char) do
-    <<lstrip(rest, char)::binary>>
+  """
+  @spec trim_leading(t, t) :: t
+  def trim_leading(string, to_trim) do
+    replace_leading(string, to_trim, "")
   end
 
-  def lstrip(string, char) when is_integer(char) do
-    string
-  end
+  @doc ~S"""
+  Returns a string where all trailing Unicode whitespaces
+  has been removed.
+
+  ## Examples
+
+      iex> String.trim_trailing("   abc\n  ")
+      "   abc"
+
+  """
+  @spec trim_trailing(t) :: t
+  defdelegate trim_trailing(string), to: String.Break
 
   @doc """
+  Returns a string where all trailing `to_trim`s have been removed.
+
+  ## Examples
+
+      iex> String.trim_trailing("_ abc __", "_")
+      "_ abc "
+
+      iex> String.trim_trailing("abc 1", "11")
+      "abc 1"
+
+  """
+  @spec trim_trailing(t, t) :: t
+  def trim_trailing(string, to_trim) do
+    replace_trailing(string, to_trim, "")
+  end
+
+  @doc ~S"""
   Returns a string where all leading and trailing Unicode whitespaces
   have been removed.
 
   ## Examples
 
-      iex> String.strip("   abc  ")
+      iex> String.trim("\n  abc\n  ")
       "abc"
 
   """
-  @spec strip(t) :: t
-
-  def strip(string) do
-    rstrip(lstrip(string))
+  @spec trim(t) :: t
+  def trim(string) do
+    string
+    |> trim_leading()
+    |> trim_trailing()
   end
 
   @doc """
-  Returns a string where all leading and trailing `char`s have been
+  Returns a string where all leading and trailing `to_trim`s have been
   removed.
 
   ## Examples
 
-      iex> String.strip("a  abc  a", ?a)
+      iex> String.trim("a  abc  a", "a")
       "  abc  "
 
   """
-  @spec strip(t, char) :: t
-  def strip(string, char) do
-    rstrip(lstrip(string, char), char)
+  @spec trim(t, t) :: t
+  def trim(string, to_trim) do
+    string
+    |> trim_leading(to_trim)
+    |> trim_trailing(to_trim)
   end
 
   @doc ~S"""
-  Returns a new string of length `len` with `subject` right justified and
-  padded with `pad`. If `pad` is not present, it defaults to
-  whitespace. When `len` is less than the length of `subject`, `subject` is
-  returned.
+  Returns a new string padded with a leading filler
+  which is made of elements from the `padding`.
+
+  Passing a string `padding` is equivalent to passing the graphemes in it.
```

Comment:
Let's add a paragraph before this one saying: "Passing a list of strings as `padding` will take one element of the list for every missing entry. If the list is shorter than the number of pads, the padding will start again from the beginning of the list."